app/src/main/java/com/example/advanceduiapp/ListActivity.kt

```diff
@@ -19,4 +21,14 @@ class ListActivity : AppCompatActivity() {
         supportActionBar?.hide()
         viewModel.imageMap.observe(this){ imageMap -> binding.imagesList.adapter = RecyclerViewAdapter(imageMap) }
     }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
```

Comment:
create some base 'ConfigurationActivity' and move that logic there instead of copy and pasting this code averywhere